modules/rgbd/src/normal.cpp

```diff
@@ -129,13 +129,22 @@ namespace rgbd
   signNormal(const Vec<T, 3> & normal_in, Vec<T, 3> & normal_out)
   {
     Vec<T, 3> res;
-    if (normal_in[2] > 0)
-      res = -normal_in / norm_vec(normal_in);
+    if(norm_vec(normal_in) == 0) { //avoid division by 0
+      normal_out[0] = 0;
```

Comment:
what I mean is: if ```norm_vec(normal_in)==0```, you are setting the normal to 0, but I think it should be NaN when giving away floats.